components/script/dom/history.rs

```diff
@@ -65,7 +64,7 @@ impl HistoryMethods for History {
         } else if delta < 0 {
             TraversalDirection::Back(-delta as usize)
         } else {
-            self.window.Location().Reload();
+            self.window.Location().reload_without_origin_check();
```

Comment:
On #whatwg they pointed out that history can't be accessed cross-origin, so once we have XOWs this point is moot. I'll put it back to `Reload`.